arch/generic/crc32_chorba_c.c

```
@@ -631,7 +637,9 @@ Z_INTERNAL uint32_t crc32_chorba_32768_nondestructive(uint32_t crc, const uint64
 }
 
 /* Implement Chorba algorithm from https://arxiv.org/abs/2412.16398 */
-Z_INTERNAL uint32_t crc32_chorba_small_nondestructive(uint32_t crc, const uint64_t *input, size_t len) {
+Z_INTERNAL uint32_t crc32_chorba_small_nondestructive(uint32_t crc, const uint8_t *input, size_t len) {
```

Cppcheck (2.19.0)

[style] The function 'crc32_chorba_small_nondestructive' should have static linkage since it is not used outside of its translation unit.

(staticFunction)